app/models/item.rb

```diff
@@ -52,6 +55,15 @@ def self.top_items_sold(merchant)
         .order("total_quantity desc")
   end
 
+  def create_slug
+    if title
+      slug = title.downcase
+      slug.gsub!(/[:@ _\&;~^`|%#?;.<>=\/\{\}\[\]\\]/, '-')
+      slug.squeeze!('-')
+      self.update(slug: slug)
```

Comment:
one thing to test here:
if a user makes an item called "Ian Was Here" and another called "Ian-was-here", the item names would be unique, but they'd end up with the same slug, which would fail on the `.update` call here. Also, there's a `.parameterize` method that would have made this a little easier: https://apidock.com/rails/ActiveSupport/Inflector/parameterize